sstables/sstables.cc

```diff
@@ -1872,6 +1878,15 @@ sstable::write_scylla_metadata(shard_id shard, sstable_enabled_features features
         _components->scylla_metadata->data.set<scylla_metadata_type::ExtTimestampStats>(std::move(*ts_stats));
     }
 
+    utils::UUID gen;
+    if (generation().is_uuid_based()) {
+        gen = generation().as_uuid();
+    } else {
+        sstlog.warn("SSTable generation {} expected to be UUID based when writing scylla metadata", generation());
```

Comment:
Why the warning? Is it an error to have non-uuid generation for an sstable?

Reply:
We are generating a unique generation here, so it may come as a surprise.
Maybe `info` level would be more appropriate.

That said, we should probably deprecate the `uuid_sstable_identifiers_enabled` option and stop supporting it as soon as possible.

Reply:
I think as long as numeric generations are valid, we should not log anything here. Why would it be a surprise that we have a numeric generation, which is still a perfectly valid option?